src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -1159,24 +1159,30 @@ internal static Type GetTypeConstraintForMethodResolution(ExpressionAst expr)
             return firstConvert == null ? null : firstConvert.Type.TypeName.GetReflectionType();
         }
 
-        internal static PSMethodInvocationConstraints CombineTypeConstraintForMethodResolution(Type targetType, Type argType)
+        internal static PSMethodInvocationConstraints CombineTypeConstraintForMethodResolution(
+            Type targetType,
+            Type argType,
+            Type[] genericArguments = null)
         {
-            if (targetType == null && argType == null)
+            if (targetType == null && argType == null && !(genericArguments?.Length > 0))
```

Comment:
Could you please make this more readable and supportable:
```suggestion
            if (targetType == null && argType == null && (genericArguments == null || genericArguments .Length == 0))
```